_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_utils.cpp

```diff
@@ -838,20 +830,46 @@ mfxStatus CopyRawSurfaceToVideoMemory(
 
         if (pSysSurface->Data.Y == 0)
         {
-            pCore->FrameAllocator.Lock(pCore->FrameAllocator.pthis, pSysSurface->Data.MemId, &lockedSurf.Data);
+            pCore->LockFrame(pSysSurface->Data.MemId, &lockedSurf.Data);
             pSysSurface = &lockedSurf;
         }
 
-        sts = pCore->CopyFrame(pCore->pthis, pDd3dSurf, pSysSurface);
+        sts = pCore->CopyFrame(pDd3dSurf, pSysSurface);
         MFX_CHECK_STS(sts);
 
         if (pSysSurface == &lockedSurf)
         {
-            pCore->FrameAllocator.Unlock(pCore->FrameAllocator.pthis, pSysSurface->Data.MemId, &lockedSurf.Data);
+            pCore->UnlockFrame(pSysSurface->Data.MemId, &lockedSurf.Data);
         }
     }
 
     return MFX_ERR_NONE;
 }
 
+mfxStatus GetNativeHandleToRawSurface(
+    VideoCORE & core,
+    mfxMemId mid,
+    mfxHDL *handle,
+    VP9MfxVideoParam const & video)
+{
+    mfxStatus sts = MFX_ERR_NONE;
+
+    mfxU32 iopattern = video.IOPattern;
+
+    mfxExtOpaqueSurfaceAlloc& opaq = GetExtBufferRef(video);
+    mfxU16 opaq_type = opaq.In.Type;
+
+    if (iopattern == MFX_IOPATTERN_IN_SYSTEM_MEMORY ||
+        (iopattern == MFX_IOPATTERN_IN_OPAQUE_MEMORY && (opaq_type & MFX_MEMTYPE_SYSTEM_MEMORY)))
+        sts = core.GetFrameHDL(mid, handle);
```

Comment:
I think you can get rid of sts variable at all and just simply return at each if statement